File: Coding/Perl/sum_product.pl
# Program that will ask a user for a number(n) and then ask if they want get the sum or product of the numbers from 1 to n

print "Hello user please enter the number n to sum from 1 to n: ";
chomp($n=<>);

print "Please enter if you want the sum(s) or the product(p) of the numbers:";
chomp($check=<>);

while($check ne 's' and $check ne 'p'){
    print "You entered an invalid value: $check. Please try again with either s or p: ";
    chomp($check=<>);
}

if($check eq 'p'){
    $sum = 1;
}
else{
    $sum = 0;
}

for($i=1;$i<=$n;$i++){
    if($check eq 's' ){
        $sum += $i;
    }
    elsif($check eq 'p'){
        $sum *= $i;
    }
    else{
        print "Incorrect input!\n"
    }
}

if($check eq 's'){
    print "The sum of the numbers 1 to n is $sum."
}
else{
    print "The product of the numbers 1 to n is $sum."
}

File: Coding/Ruby/sum_product.rb
# Program to get a number n from the user and then asks if the user wants the sum or product of the numbers 1 to n

puts "Hello user please enter the number n: "
n = gets.to_i

puts "Please enter either sum(s) or product(p): "
check = gets.chomp

while (check != 's' and check != 'p')
    puts "Invalid input: #{check}. Please enter a s or p."
    check = gets.chomp
end

if (check == 'p')
    sum = 1
else 
    sum = 0
end

for i in 1 .. n
    if check == 's'
        sum += i
    elsif check == 'p'
        sum *= i
    else
        puts "Invalid input!"
    end
end

if (check == 's')
    puts "The sum of the numbers 1 to #{n} is #{sum}."
else
    puts "The product of the numbers from 1 to #{n} is #{sum}."
end